dora/core/server/master/src/main/java/alluxio/master/scheduler/Scheduler.java

```diff
@@ -396,13 +406,103 @@ public JobMetaStore getJobMetaStore() {
     return mJobMetaStore;
   }
 
+  /**
+   * Bounded priority queue impl.
+   * @param <E>
+   */
+  @SuppressFBWarnings({"SE_BAD_FIELD_INNER_CLASS", "SE_NO_SERIALVERSIONID"})
+  public class BoundedPriorityBlockingQueue<E> extends PriorityBlockingQueue<E> {
```

Comment:
Feel like we should have a mechanism to say that we have too many tasks in the worker queue so we don't pull any tasks anymore. otherwise, the memory would be growing if we keep adding tasks to the retryTasksList in the job. A simple comparison with MAX_TASK_PER_WORKER*NUM_WORKERS could do the work

Reply:
each q for each worker is bounded in size by MAX_TASK_PER_WORKER